### DISCUSSION TITLE
Data modeling (one-to-many and many-to-many)

### DISCUSSION THREAD

**Discussion by brunocrosier** [supabase/supabase#1681] [Questions] [answered]
I'm trying out Supabase and can't seem to get my head around how to properly model data using the GUI

I'm trying out some simple relationships such as "one book has many reviews"

Am I going about this the wrong way?

Here's what happens:

https://user-images.githubusercontent.com/18399089/119233211-cc3f1a80-bb1f-11eb-8ad8-93ac5ebad902.mp4

Error message seems to be: `foreign key constraint "books_reviews_fkey" cannot be implemented`

Any help appreciated!

**Comment by soedirgo** [comment, marked as answer]
Hey @brunocrosier, this is because you can't have an array of foreign keys (see [here](https://stackoverflow.com/questions/41054507/postgresql-array-of-elements-that-each-are-a-foreign-key) for explanation). Instead you might want to consider having a `book_id` column in `reviews` that references `books.id`.

**Reply by brunocrosier** [reply]
Thank you! What about for a many-to-many relationship? Would we have to create a bridging / junction table?

**Reply by soedirgo** [reply]
Yup! You can see an example for that here https://github.com/supabase/supabase/discussions/710#discussioncomment-362651.

**Reply by brunocrosier** [reply]
Ahh you're a star! Thanks for that. Hope they add something like that to the docs soon!

**Reply by Cardoso-topdev** [reply]
I have a question!
After adding a `book_id `column in `reviews` table that references `books.id`, how can I get all `reviews` from `books` table?
Is it possible?

**Reply by soedirgo** [reply]
@Cardoso-topdev you can try the "Query foreign tables" example [here](https://supabase.com/docs/reference/javascript/select), replacing `countries` with `books` and `cities` with `reviews`.

**Reply by Cardoso-topdev** [reply]
@soedirgo ! Thank you for your kind answer.
One more question, could you tell me some example for many to many relations, please?
I tried to follow `Query foreign tables through a join table` case, but it's response empty array.
Do you know the way point out the relation table with pointing out the exact field names?
Here is a screen shot of my tables with relations.
![image](https://user-images.githubusercontent.com/75736745/221196839-8b123ee8-1414-4987-b912-f1979e5df304.png)

I used this code, but the response's topic is an empty array([]). 
```
await supabase
  .from('posts')
  .select(`*,
    topics (
      *
    )
  `);
```
Thank you.

**Reply by soedirgo** [reply]
You can check out the link above: https://github.com/supabase/supabase/discussions/710#discussioncomment-362651

I don't have enough info to deduce why it's an empty array, but you can try doing `select('*, posts_topics(*)'` to see if it's still an empty array.

**Reply by Cardoso-topdev** [reply]
@soedirgo ! 
I solved the problem by adding a new policy for the new relation table(posts_topics).
I forgotten to add a new policy for it.
Thank you very much.
